src/main/scala/mesosphere/marathon/core/auth/AuthModule.scala

```diff
@@ -0,0 +1,25 @@
+package mesosphere.marathon.core.auth
+
+import mesosphere.marathon.WrongConfigurationException
+import mesosphere.marathon.core.auth.impl.AuthAllowEverything
+import mesosphere.marathon.core.plugin.PluginManager
+import mesosphere.marathon.plugin.auth.{ Authenticator, Authorizer }
+
+import scala.reflect.ClassTag
+
+class AuthModule(pluginManager: PluginManager, authConfig: AuthConfig) {
+
+  private[this] def plugin[T](className: String)(implicit ct: ClassTag[T]): T = {
+    pluginManager.plugin[T](className).getOrElse(
+      throw WrongConfigurationException(s"No ${ct.runtimeClass.getSimpleName} with className $className found!")
```

Comment:
nit:

s"${ct.runtimeClass.getSimpleName} with className $className not found!" sounds better to me.